index.js

```diff
@@ -17,14 +18,31 @@ require('react-native').unstable_enableLogBox();
 
 enableScreens();
 
-const Elegant = () => (
-  <Provider store={store}>
-    <PersistGate loading={<Loading />} persistor={persistor}>
-      <PaperProvider>
-        <App />
-      </PaperProvider>
-    </PersistGate>
-  </Provider>
-);
+const Elegant = () => {
+  const [theme, setTheme] = React.useState('light');
+
+  function toggleTheme() {
+    setTheme((theme) => (theme === 'light' ? 'dark' : 'light'));
+  }
```

Comment:
I guess you should worry about re-rendering and re-making of the `toggleTheme` component. I suggest you use `useCallback` hook function.